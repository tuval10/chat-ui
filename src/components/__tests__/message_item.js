import React from 'react';
import {create} from 'react-test-renderer';
import MessageItem from '../message_item';
import {user as myUser, message, otherUserMessage} from '../../test_helpers/fixtures'

describe('MessageItem component', () => {
  describe('renders correctly', () => {
    it('when message belongs to user', () => {
      const tree = create(
        <MessageItem {...{myUser, ...message}} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('when message does not belong to user', () => {
      const tree = create(
          <MessageItem {...{myUser, ...otherUserMessage}} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});