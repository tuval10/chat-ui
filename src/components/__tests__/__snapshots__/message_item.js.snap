// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem component renders correctly when message belongs to user 1`] = `
<div
  className="message-item mine"
>
  <div
    className="message-details"
  >
    <div
      className="avatar-container"
    >
      <img
        alt="user-logo"
        className="avatar-icon"
        src="https://spotim-demo-chat-server.herokuapp.com/avatars/005-bullbasaur.png"
      />
    </div>
    <div
      className="username"
    >
      stub_username
    </div>
    <div
      className="posted-at"
    >
      00:34
    </div>
  </div>
  <div
    className="message-text"
  >
    stub message
  </div>
</div>
`;

exports[`MessageItem component renders correctly when message does not belong to user 1`] = `
<div
  className="message-item"
>
  <div
    className="message-details"
  >
    <div
      className="avatar-container"
    >
      <img
        alt="user-logo"
        className="avatar-icon"
        src="https://spotim-demo-chat-server.herokuapp.com/avatars/004-jigglypuff.png"
      />
    </div>
    <div
      className="username"
    >
      stub_username 2
    </div>
    <div
      className="posted-at"
    >
      00:35
    </div>
  </div>
  <div
    className="message-text"
  >
    stub message
  </div>
</div>
`;
